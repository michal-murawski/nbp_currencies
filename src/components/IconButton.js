import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import IconButtonMUI from 'material-ui/IconButton';

const styles = theme => ({
  root: {
    width: 30,
    height: 30,
    fontSize: '1.2rem',
  },
});

function IconButton({ children, classes, onClick }) {
  return (
    <IconButtonMUI classes={{ root: classes.root }} onClick={onClick}>
      {children}
    </IconButtonMUI>
  );
}

IconButton.propTypes = {
  onClick: PropTypes.func,
  classes: PropTypes.object,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
    .isRequired,
};

export default withStyles(styles)(IconButton);
