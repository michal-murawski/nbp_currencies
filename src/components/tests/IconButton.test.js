import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import IconButton from '../IconButton';

const mockOnClick = jest.fn();
const mockClasses = {
  root: 'test_class',
};
describe('components/IconButton', () => {
  let wrapper;
  const ChildElement = () => <div>Test value</div>;

  beforeEach(() => {
    wrapper = shallow(
      <IconButton onClick={mockOnClick} classes={mockClasses}>
        <ChildElement />
      </IconButton>
    );
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should contain ChildElement and proper class', () => {
    expect(wrapper.find(ChildElement)).toHaveLength(1);
    expect(wrapper.props().classes.root.includes(mockClasses.root)).toBe(true);
  });

  it('should call onClick method', () => {
    expect(mockOnClick).toHaveBeenCalledTimes(0);

    wrapper.simulate('click');

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
