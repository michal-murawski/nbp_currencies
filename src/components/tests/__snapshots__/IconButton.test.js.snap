// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/IconButton should render properly 1`] = `
<IconButton
  classes={
    Object {
      "root": "IconButton-root-1 test_class",
    }
  }
  onClick={[Function]}
>
  <ChildElement />
</IconButton>
`;
